daemon/cmd/cni.go

```diff
@@ -125,39 +131,87 @@ func (d *Daemon) startCNIConfWriter(opts *option.DaemonConfig, cleaner *daemonCl
 		return
 	}
 
+	cniConfDir, _ := cniConfPaths(opts)
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Warnf("Failed to create watcher: %v", err)
```

Comment:
in this case, we should watcher to be explicitly nil in case NewWatcher returns something other than nil on its error condition (or decides to in the future).

Reply:
Done